addons/hw_posbox_homepage/controllers/main.py

```diff
@@ -90,13 +90,15 @@ def get_homepage_data(self):
 
         pos_device = self.get_pos_device_status()
         iot_device = []
-        for status in pos_device:
-            if pos_device[status]['status'] == 'connected':
-                iot_device.append({
-                    'name': status,
-                    'type': 'device',
-                    'message': ' '.join(pos_device[status]['messages'])
-                })
+
+        if not iot_devices:
```

Comment:
if not here, not in the loop after

Reply:
Sorry I don't understand. I don't see any other loop on pos_device.